pkg/common-controller/snapshot_controller.go

```diff
@@ -149,31 +149,6 @@ func (ctrl *csiSnapshotCommonController) syncContent(content *crdv1.VolumeSnapsh
 			ctrl.snapshotQueue.Add(snapshotName)
 		}
 	}
-
-	// NOTE(xyang): Do not trigger content deletion if
-	// snapshot is nil. This is to avoid data loss if
-	// the user copied the yaml files and expect it to work
-	// in a different setup. In this case snapshot is nil.
-	// If we trigger content deletion, it will delete
-	// physical snapshot resource on the storage system
-	// and result in data loss!
-	//
-	// Trigger content deletion if snapshot is not nil
-	// and snapshot has deletion timestamp.
-	// If snapshot has deletion timestamp and finalizers, set
-	// AnnVolumeSnapshotBeingDeleted annotation on the content.
-	// This may trigger the deletion of the content in the
-	// sidecar controller depending on the deletion policy
-	// on the content.
-	// Snapshot won't be deleted until content is deleted
-	// due to the finalizer.
```

Comment:
What is the reason to remove this whole block?

Reply:
https://github.com/kubernetes-csi/external-snapshotter/blob/3a35a5e48640085658d07b61eea3e0bdc3a757c9/pkg/common-controller/snapshot_controller.go#L332-L337
This code should already handle it.

In the commit message of second commit, I wrote:
> The original annotate logic in syncContent() is not needed because all deleting snapshot will go through above logic. And natually, if a content is never bound, is will never get the vs-deleted annotation.

Reply:
Hi @xing-yang What do you think about the overall direction of this PR? Should we continue to work on this? These change is not required for this PR. If you agree, I can defer it to another PR.